### PR TITLE
Add linting + formatting

### PR DESCRIPTION
**Pull request by dsiddharth**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,8 +47,12 @@
     "@types/update-notifier": "5.1.0",
     "@types/uuid": "8.3.4",
     "@types/yargs": "17.0.10",
+    "@typescript-eslint/eslint-plugin": "^5.25.0",
+    "@typescript-eslint/parser": "^5.25.0",
+    "eslint": "^8.16.0",
+    "eslint-plugin-import": "^2.26.0",
     "prettier": "2.6.2",
-    "typescript": "4.6.4"
+    "typescript": "^4.6.4"
```
Comment (hpx7) [inline package.json]: can we get rid of the `^` on these? (is there a lint rule for that?)

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 node_modules/
 .DS_Store
 .idea
+.vscode/
```
Comment (hpx7) [inline .gitignore]: add newline

File: package.json (hunk 1 above)
Comment (dsiddharth) [reply package.json]: Figured it out -- take a look at the newer commits.

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+examples/
+.eslintrc.*
```
Comment (hpx7) [inline .eslintignore]: newline

**Review by hpx7 — APPROVED** [review]

**Pull request merged**